modules/ServerAPI/src/server/routes/ProtocolRoutes.coffee

```diff
@@ -0,0 +1,97 @@
+exports.setupRoutes = (app, loginRoutes) ->
+	app.post '/api/getTemplateSELFile', loginRoutes.ensureAuthenticated, exports.getTemplateSELFile
+
+exports.getTemplateSELFile = (req, resp) ->
+    config = require '../conf/compiled/conf.js'
+    experimentServiceRoutes = require './ExperimentServiceRoutes.js'
+    
+    protocolCode = req.body.protocolCode
+    protocolName = req.body.protocolName
+    protocolScientist = req.body.protocolScientist
+    protocolDate = req.body.protocolDate
+    protocolProject = req.body.protocolProject
+
+    endpointNames = req.body.endpointData.endpointNames
+    endpointUnits = req.body.endpointData.endpointUnits
+    endpointDataTypes = req.body.endpointData.endpointDataTypes
+    endpointConc = req.body.endpointData.endpointConc
+    endpointConcUnits = req.body.endpointData.endpointConcUnits
+    endpointTime = req.body.endpointData.endpointTime
+    endpointTimeUnits = req.body.endpointData.endpointTimeUnits
+    endpointHidden = req.body.endpointData.endpointHidden
+
+    try
+        # create a CSV file with the endpoints	
+        blankElements = ["NA", "undefined", "", null, undefined]
+
+        endpointNameRowString = "Corporate Batch ID,"
+        dataTypeRowString = "Datatype,"
+        for indexNum in [0..endpointNames.length]
+            endpointRowEntry = ""
+            dataTypeEntry = ""
+
+            endpointHasNoValues = true
+
+            if endpointNames[indexNum] not in blankElements
```

Comment:
nit: IMO it'd be more readable if you just set the endpointName, unit, conc, etc. to variables up front:
```
endpointName = endpointNames[indexNum]
endpointUnit = endpointUnits[indexNum]
endpointConc[indexNum]
...
```

Reply:
Done!